wrangler-core/src/main/java/io/cdap/directives/datetime/DateTimeToTimeStamp.java

```diff
@@ -0,0 +1,116 @@
+ *  License for the specific language governing permissions and limitations under
+ *  the License.
+ */
+package io.cdap.directives.datetime;
+
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.wrangler.api.Arguments;
+import io.cdap.wrangler.api.Directive;
+import io.cdap.wrangler.api.DirectiveParseException;
+import io.cdap.wrangler.api.ErrorRowException;
+import io.cdap.wrangler.api.ExecutorContext;
+import io.cdap.wrangler.api.Optional;
+import io.cdap.wrangler.api.Row;
+import io.cdap.wrangler.api.annotations.Categories;
+import io.cdap.wrangler.api.lineage.Lineage;
+import io.cdap.wrangler.api.lineage.Mutation;
+import io.cdap.wrangler.api.parser.ColumnName;
+import io.cdap.wrangler.api.parser.TokenType;
+import io.cdap.wrangler.api.parser.UsageDefinition;
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.zone.ZoneRulesException;
+import java.util.List;
+
+/**
+ * Directive for converting a datetime column to timestamp with the specified zone
+ */
+@Plugin(type = Directive.TYPE)
+@Name("timestamp")
+@Categories(categories = {"datetime"})
+@Description("Converts a datetime column to timestamp")
+public class DateTimeToTimeStamp implements Directive, Lineage {
+
+  public static final String NAME = "timestamp";
+  private static final String COLUMN = "column";
+  private static final String ZONE = "timezone";
+  private String column;
+  private String zone;
+  private ZoneId zoneId;
+
+  @Override
+  public UsageDefinition define() {
+    UsageDefinition.Builder builder = UsageDefinition.builder(NAME);
+    builder.define(COLUMN, TokenType.COLUMN_NAME);
+    builder.define(ZONE, TokenType.TEXT, Optional.TRUE);
+    return builder.build();
+  }
+
+  @Override
+  public void initialize(Arguments args) throws DirectiveParseException {
+    this.column = ((ColumnName) args.value(COLUMN)).value();
+    if (args.value(ZONE) == null) {
+      this.zoneId = ZoneId.of("UTC");
+      this.zone = this.zoneId.toString();
+      return;
+    }
+    this.zone = args.value(ZONE).value().toString();
+    try {
+      this.zoneId = ZoneId.of(this.zone);
+    } catch (IllegalArgumentException | ZoneRulesException exception) {
+      throw new DirectiveParseException(NAME, String.format("Zone '%s' is invalid.", this.zone), exception);
+    }
+  }
+
+  @Override
+  public List<Row> execute(List<Row> rows, ExecutorContext context) throws ErrorRowException {
+    for (Row row : rows) {
+      int idx = row.find(column);
+      if (idx == -1) {
+        continue;
+      }
+      Object value = row.getValue(idx);
+      // If the data in the cell is null, then skip this row.
+      if (value == null) {
+        continue;
+      }
+
+      if (!(value instanceof LocalDateTime)) {
+        throw new ErrorRowException(NAME, String.format("Column %s is not of DateTime type", value.toString()), 2);
```

Comment:
Is this supposed to be the column name instead of the value?

Also would be good to mention what the type actually is, something like:

Expected column x to be a DateTime but found a y

Reply:
Done